src/MongoDB.EntityFrameworkCore/Storage/MongoDatabaseWrapper.cs

```diff
@@ -217,4 +148,148 @@ private static string GetId(IUpdateEntry entry)
             _ => throw new NotSupportedException($"Unknown type for _id property on entity type ${entry.EntityType.DisplayName()}")
         };
     }
+
+    private IEnumerable<MongoUpdate> ConvertUpdateEntriesToMongoUpdates(IEnumerable<IUpdateEntry> entries)
+    {
+        return
+            GetAllChangedRootEntries(entries)
+            .Select(ConvertUpdateEntryToMongoUpdate)
+            .Where(x => x != null)
+            .Cast<MongoUpdate>();
```

Comment:
The `Cast` is to shut up a compiler warning about nullability. Apparently the compiler doesn't realize that `Where(x => x != null)` guarantees there won't be any null values.

Reply:
tip: you can shortcut both lines into just `OfType`
```diff
    .Select(ConvertUpdateEntryToMongoUpdate)
+    .OfType<MongoUpdate>();
-    .Where(x => x != null)
-    .Cast<MongoUpdate>();
```